packages/qwik/src/core/component/component.unit.tsx

```diff
@@ -156,8 +157,44 @@ describe('q-component', () => {
             value="1"
           />
           <Input2 value="2" />
-          <Input5 value="5" type="text" />
-          <Input6 value="6" type="number" />
+          <Input3 value="3" type="text" />
+          <Input4 value="4" type="number" />
+        </>
+      );
+    });
+  });
+
+  test('custom PropFunction types should work', () => () => {
+    type TestProps = PropsOf<'h1'> & {
+      test$?: () => void;
+    };
+    const Test = component$<TestProps>(({ test$, ...props }) => {
+      return (
+        <>
+          <h1 onClick$={() => test$} {...props}>
```

Comment:
this should be `onClick$={test$}`, no? And you also need to do some type assertions like in jsx-types.unit.tsx